src/db-adapters/Mongoose/MongooseAdapter.js

```diff
@@ -614,4 +592,27 @@ export default class MongooseAdapter {
 
     return words.join(" ");
   }
+
+  static makeIdQuery(idOrIds) {
+    const idIsValid = mongoose.Types.ObjectId.isValid;
+    let idQuery;
+
+    if (typeof idOrIds === "string") {
+      if (!idIsValid(idOrIds)) {
+        throw new APIError(404, undefined, "No matching resource found.");
+      }
+
+      idQuery = idOrIds;
+    } else if (Array.isArray(idOrIds)) {
+      let validIds = idOrIds.filter(idIsValid);
```

Comment:
My thinking here was that this validation step is purely to prevent Mongoose producing casting errors, and therefore an invalid ID should give the same result as a valid but missing ID.